framework/src/bundle/BundleContext.cpp

```diff
@@ -74,12 +74,20 @@ BundleContext& BundleContext::operator=(std::nullptr_t)
 
 std::shared_ptr<detail::LogSink> BundleContext::GetLogSink() const
 {
+  if (!d) {
+    throw std::runtime_error("The bundle context is no longer valid");
+  }
+
   d->CheckValid();
   return d->bundle->coreCtx->sink->shared_from_this();
 }
 
 Any BundleContext::GetProperty(const std::string& key) const
 {
+  if (!d) {
+    throw std::runtime_error("The bundle context is no longer valid");
```

Comment:
The doxygen comment in `BundleContext.h` needs to be updated to indicate this now throws.